src/charm.py

```diff
@@ -317,7 +317,7 @@ def _should_run_setup(self, current_plan: Dict, s3info: Optional[S3Info]) -> boo
 
         Returns:
             If no services are planned yet (first run) or S3 settings have changed.
-        """
+        """  # pylint: disable=line-too-long
```

Comment:
I'm guessing the offending line is the first one, should we instead summarise it and add a more detailed description in the body? E.g.;

```Python
        """Determine if the setup script is to be run.

        This is based on the current plan and the new S3 settings.

        Args:
            current_plan: Dictionary containing the current plan.
            s3info: S3Info object containing the S3 configuration options.

         Returns:
             If no services are planned yet (first run) or S3 settings have changed.
         """
```

Reply:
Definitely better, thanks!